### DISCUSSION TITLE
32-bit/x86 Windows 10 install possible?

### DISCUSSION THREAD

**Discussion by zhudnall** [utmapp/UTM#6370] [Windows]
I have tried to install every Windows 10 32-bit/x86 version I could find, and with UTM at the most I get is either freeze on disc boot or the initial "waiting" circle of death hang, never gets to install screen.

I've tried multiple configurations, x86_64, x86, amd64... I've tried Intel emulation, qemu64 and qemu32, default. I've tried 2009 and 1996 machines.

I have done UEFI/EFI and non-UEFI boot options. 

I'm running out of options and I was wondering if anyone has had any success with this, because I'm not seeing much in the Issues log.

UTM Version: 4.4.5
macOS Version: 14.4.1 (23E224)
Mac Chip (Intel, M1, ...):  M3 Pro

**If you have that persistent question of "bUT wHy?!"... its 32-bit legacy software / drivers, because why the hell else would I need to subject myself to this torture.

**Comment by osy** [comment]
Please provide your config.plist

**Reply by zhudnall** [reply]
osy,

Thanks for your reply. My question is are there ANY configuration that works with Windows x86 architecture? I have used  multiple configurations with minimal progress. It would get to a Windows installer boot loop and never progress further.

I'm looking for someone that has managed to successfully run a 32bit Windows 10 instance with UTM in it's current form. And if so, how was it done and what was the configuration?

**Reply by osy** [reply]
Everything you said should work that's why I'm asking for your configuration

**Reply by zhudnall** [reply]
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>Backend</key>
	<string>QEMU</string>
	<key>ConfigurationVersion</key>
	<integer>4</integer>
	<key>Display</key>
	<array>
		<dict>
			<key>DownscalingFilter</key>
			<string>Linear</string>
			<key>DynamicResolution</key>
			<true/>
			<key>Hardware</key>
			<string>VGA</string>
			<key>NativeResolution</key>
			<false/>
			<key>UpscalingFilter</key>
			<string>Nearest</string>
		</dict>
	</array>
	<key>Drive</key>
	<array>
		<dict>
			<key>Identifier</key>
			<string>5CD9E4F5-C840-4BAD-8AD2-F26B5DC70BF6</string>
			<key>ImageType</key>
			<string>CD</string>
			<key>Interface</key>
			<string>IDE</string>
			<key>InterfaceVersion</key>
			<integer>1</integer>
			<key>ReadOnly</key>
			<true/>
		</dict>
		<dict>
			<key>Identifier</key>
			<string>F7845FB0-003E-48B8-9AA8-BC3CA6C595E5</string>
			<key>ImageName</key>
			<string>F7845FB0-003E-48B8-9AA8-BC3CA6C595E5.qcow2</string>
			<key>ImageType</key>
			<string>Disk</string>
			<key>Interface</key>
			<string>IDE</string>
			<key>InterfaceVersion</key>
			<integer>1</integer>
			<key>ReadOnly</key>
			<false/>
		</dict>
		<dict>
			<key>Identifier</key>
			<string>1ECDF945-7EF4-4B05-A85E-44DE225619DB</string>
			<key>ImageType</key>
			<string>CD</string>
			<key>Interface</key>
			<string>IDE</string>
			<key>InterfaceVersion</key>
			<integer>1</integer>
			<key>ReadOnly</key>
			<true/>
		</dict>
	</array>
	<key>Information</key>
	<dict>
		<key>Icon</key>
		<string>windows</string>
		<key>IconCustom</key>
		<false/>
		<key>Name</key>
		<string>Windows 10 (32bit)</string>
		<key>UUID</key>
		<string>BA9015F0-B02A-4AC6-9019-B881B7C864AD</string>
	</dict>
	<key>Input</key>
	<dict>
		<key>MaximumUsbShare</key>
		<integer>3</integer>
		<key>UsbBusSupport</key>
		<string>3.0</string>
		<key>UsbSharing</key>
		<false/>
	</dict>
	<key>Network</key>
	<array>
		<dict>
			<key>Hardware</key>
			<string>rtl8139</string>
			<key>IsolateFromHost</key>
			<false/>
			<key>MacAddress</key>
			<string>F2:F3:9A:C7:0E:63</string>
			<key>Mode</key>
			<string>Shared</string>
			<key>PortForward</key>
			<array/>
		</dict>
	</array>
	<key>QEMU</key>
	<dict>
		<key>AdditionalArguments</key>
		<array/>
		<key>BalloonDevice</key>
		<false/>
		<key>DebugLog</key>
		<false/>
		<key>Hypervisor</key>
		<false/>
		<key>PS2Controller</key>
		<false/>
		<key>RNGDevice</key>
		<false/>
		<key>RTCLocalTime</key>
		<false/>
		<key>TPMDevice</key>
		<false/>
		<key>TSO</key>
		<false/>
		<key>UEFIBoot</key>
		<false/>
	</dict>
	<key>Serial</key>
	<array/>
	<key>Sharing</key>
	<dict>
		<key>ClipboardSharing</key>
		<true/>
		<key>DirectoryShareMode</key>
		<string>WebDAV</string>
		<key>DirectoryShareReadOnly</key>
		<false/>
	</dict>
	<key>Sound</key>
	<array>
		<dict>
			<key>Hardware</key>
			<string>AC97</string>
		</dict>
	</array>
	<key>System</key>
	<dict>
		<key>Architecture</key>
		<string>x86_64</string>
		<key>CPU</key>
		<string>qemu64-v1</string>
		<key>CPUCount</key>
		<integer>4</integer>
		<key>CPUFlagsAdd</key>
		<array/>
		<key>CPUFlagsRemove</key>
		<array/>
		<key>ForceMulticore</key>
		<true/>
		<key>JITCacheSize</key>
		<integer>0</integer>
		<key>MemorySize</key>
		<integer>4096</integer>
		<key>Target</key>
		<string>pc</string>
	</dict>
</dict>
</plist>

```

**Reply by osy** [reply]
Try changing the CPU to Broadwell-v4 or SandyBridge-v2 or something not too old but not too new

**Reply by zhudnall** [reply]
I had the same thought process for the emulation layer. I already attempted Broadwell because it was in fact the last Intel chip I've personally purchased.  I also had no luck with the SandyBridge, Haswell, or IvyBridge. Neither with Default or Qemu Athlon/32/64.

I always get stuck in the same spot after boot

<img width="1020" alt="Screenshot 2024-05-22 at 7 10 07 PM" src="https://github.com/utmapp/UTM/assets/22178764/c1fc3879-abef-4a3f-9430-ce07eae99551">

**Reply by osy** [reply]
Maybe turn off force multicore?

**Reply by zhudnall** [reply]
I tried removing force multicore, and also tried setting number of cores to default. I also tested across all the previous stated processors with no avail.

**Reply by zhudnall** [reply]
Also for the sake of completeness, I am attempting to emulate the Windows 10 x32 22h2 version from Microsoft.

**Comment by C4J** [comment]
I also have this problem. I am running on a MacBook Apple Silicon M3.

I have got Windows 11 and Windows XP working with UTM emulation - but Windows 10 (32bit) just won't install. I have download the media via CrystalFetch and from Microsoft direct but when I boot the new VM is just goes into the EFI boot shell and does not even try to boot from the DVD. I have tried the above suggestions without success. I did not have any issues with Win 11 and Windows XP. Very odd.

**Comment by zhudnall** [comment]
I noticed that this was moved from Issues to Discussions... I'm not sure that was the correct move.
